orderer/common/multichannel/registrar.go

```diff
@@ -430,8 +450,42 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block) (types.
 		return types.ChannelInfo{}, types.ErrChannelAlreadyExists
 	}
 
-	//TODO
-	return types.ChannelInfo{}, errors.New("Not implemented yet")
+	if !isAppChannel && len(r.chains) > 0 {
+		return types.ChannelInfo{}, types.ErrAppChannelsAlreadyExists
```

Comment:
shouldn't this be "system channel already exists" ?

Reply:
No, This is for when a user is trying to create a system channel (`isAppChannel=false`), but the  orderer is already working with app. channel only mode (i.e., with participation API). This not supported. We only (will) allow the creation of the system channel wit a join when there are no channels at all.